Assignment_04_09.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Sep  4 09:40:58 2021
+
+@author: akhom4549
+"""
+
+
+import pandas as pd
+import numpy as np
+import seaborn as sns
+from scipy.stats import levene
+
+
+df1 = pd.read_csv(r"C:\Users\akhom4549\Desktop\Training\bike_sharing_hours.csv")
+
+
+# Ques1 solution
+
+df1['dteday'] = pd.to_datetime(df1['dteday'])
+df1['day'] = df1['dteday'].dt.day
+levene(df1.loc[(df1['mnth']<=6)&(df1['day']<=7),'cnt'],df1.loc[(df1['mnth']>=7)&(df1['day']<=7),'cnt'])
+
+# statistically significant since pvalue is 2.30E-08 hence we say the variances are not equal
+
+from scipy.stats import ttest_ind
+x = ttest_ind(df1.loc[(df1['mnth']<=6)&(df1['day']<=7),'cnt'],
+          df1.loc[(df1['mnth']>=7)&(df1['day']<=7),'cnt'],
+          equal_var=False)
+
+# p value is 9.4E-20 hence we say the average number of rented bikes in first 7 days differ significantly
+
+# average temperature in first quarter of the year taking first quarter as calendar quarter (Jan'-Mar)
+df1[df1['mnth']<=3]['temp'].mean()
+# this returns mean as 0.3106245
+
+from scipy.stats import ttest_1samp
+ttest_1samp(df1[(df1['mnth']>=7)&(df1['mnth']<=9)]['temp'],popmean=0.3106245)
+
+# 0 p value is less than 0.05 hence we reject the null hypothesis, 
+# hence we say that temperature has changed significantly
+
+# checking for validation
+df1[(df1['mnth']>=7)&(df1['mnth']<=9)]['temp'].mean()
+
+from scipy.stats import chi2_contingency
+k=pd.crosstab(df1['casual'],df1['weekday']).values
+res=chi2_contingency(k)
+res[1]
+
```

Comment:
casual is not categorical feature . this should have been anova , not chisquared test